source/administration/plugins.rst

```diff
@@ -4,148 +4,96 @@ Plugins (Beta)
+While this power enables deep customization and integration, it can be abused in the wrong hands. Plugins have full access to your server configuration and thus your Mattermost database. Plugins can read any message in any channel, or perform any action on behalf of any user in the webapp.
 
-Web App Considerations
-~~~~~~~~~~~~~~~~~~~~~~
-Plugins have the ability to execute arbitrary code in client browsers. They have the ability to perform nearly any action on behalf of anyone using the web app.
+You should only install custom plugins from sources you trust to avoid compromising the security of your installation.
 
 Set Up Guide
---------------
-
-To enable plugins, go to **System Console > Plugins > Plugin Management** in versions after 5.12, or follow these two steps in prior versions:
-
-1) Go to **System Console > Plugins (Beta) > Configuration**. Here you can enable plugins.
-2) Go to **System Console > Plugins (Beta) > Management** to manage your plugins, including:
-
- - Activating or deactivating pre-packaged Mattermost plugins.
- - Upload and install a plugin for your Mattermost server, or remove them.
+------------
 
-For more information on building your own plugin, see our `developer documentation <https://developers.mattermost.com/extend/plugins/>`__.
+To manage plugins, go to **System Console > Plugins > Plugin Management**. Fom here, you can:
 
-Plugin Uploads
-~~~~~~~~~~~~~~~~~~
+ - Enable or disable prepackaged plugins.
+ - Install and manage custom plugins
 
-Mattermost supports plugin uploads by System Admins, which allow you to customize and extend the platform in ways that would otherwise not be available. These plugins are not pre-packaged in Mattermost, and have either been developed by the community or by a Mattermost staff member.
-
-By default, plugin uploads are disabled on your server. To enable them, set **PluginSettings > EnableUploads** to ``true`` in your ``config.json`` file. You can disable plugin uploads anytime to control which plugins are installed on your server. This action won't disable plugins already installed on your server.
+.. note::
+  In prior versions, go to **System Console > Plugins (Beta) > Configuration**.
 
-Once enabled, install plugins in one of the following ways. The steps assume you previously generated a ``plugin.tar.gz`` file.
 
-1) Through System Console UI:
- - Log in to Mattermost as a System Admin.
- - Navigate to **Plugins > Management** and upload ``plugin.tar.gz``.
- - Click "Activate" under the plugin after it has uploaded.
+Prepackaged Plugins
+~~~~~~~~~~~~~~~~~~~
+Mattermost ships with a number of prepackaged plugins written and maintained by Mattermost. Instead of building these features directly into the product, you can selectively enable the functionality your installation requires. Prepackaged plugins cannot be removed via the System Console, but can be customized by modifying the `prepackaged_plugins` directory in your Mattermost installation.
```

Comment:
```suggestion
Mattermost ships with a number of prepackaged plugins written and maintained by Mattermost. Instead of building these features directly into the product, you can selectively enable the functionality your installation requires. Prepackaged plugins cannot be removed via the System Console, but can be customized by modifying the ``prepackaged_plugins`` directory in your Mattermost installation.
```